src/nvim/quickfix.c

```diff
@@ -792,24 +895,33 @@ static bool qf_stack_empty(const qf_info_T *qi)
 }
 
 // Returns true if the specified quickfix/location list is empty.
-static bool qf_list_empty(const qf_info_T *qi, int qf_idx)
+static bool qf_list_empty(qf_list_T *qfl)
   FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
 {
-  if (qi == NULL || qf_idx < 0 || qf_idx >= LISTCOUNT) {
-    return true;
-  }
-  return qi->qf_lists[qf_idx].qf_count <= 0;
+  return qfl == NULL || qfl->qf_count <= 0;
+}
+
+// Returns TRUE if the specified quickfix/location list is not empty and
+// has valid entries.
+static int qf_list_has_valid_entries(qf_list_T *qfl)
```

Comment:
Opposed to Vim, we use C99. That means we can use the `bool` type as well. This should be `static bool`. Then the comment above should be changed from `TRUE` to `true`.

On compiling you will probably get quite a few warnings that other binary variables etc. using this function should be changed from `int` to `bool` as well.